docs/core/tools/dotnet-nuget-push.md

```diff
@@ -148,5 +149,25 @@ The command pushes an existing package. It doesn't create a package. To create a
   ```dotnetcli
   dotnet nuget push "*.nupkg" -s c:\mydir
   ```
+  
+- In the example below pushing "Foo" version "5.0.2" to Azure DevOps Artifacts from dev box, here AZ is just a placeholder for ApiKey, this prevents authentication fail prematurely, in order to authentication to work you need to install [cred provider](https://github.com/microsoft/artifacts-credprovider). Below command trigger open Cred Provider window if authentication is necessary, it's suitable for pushing from dev box, but not for CI.
```

Comment:
Similarly to above, I think we should have a dedicated section for azure devops.

Reply:
Feels like we should be referencing an azure devops page with instructions for this instead.

Does Azure Devops have documentation on how to handle pushing at all?

Reply:
AzDO does have this page on publishing packages: https://docs.microsoft.com/en-us/azure/devops/pipelines/artifacts/nuget?toc=%2Fazure%2Fdevops%2Fartifacts%2Ftoc.json&bc=%2Fazure%2Fdevops%2Fartifacts%2Fbreadcrumb%2Ftoc.json&view=azure-devops&tabs=yaml


I don't feel 100% comfortable putting feed specific instructions in our push docs that's generic for all feeds. We certainly do have a lot of Azure Artifacts customers, and it is a Microsoft service. And if customers come to this help page, but don't find Azure Artifacts' docs page, then instructions here can help them be successful.

On the other hand, GitHub and nuget.org and are also owned by Microsoft. Should we put instructions for those NuGet servers in this doc too?  How about non-Microsoft servers like MyGet, GitLabs, TeamCity, Artifactory, and who knows how many more?  I might be just too idealistic, but I don't feel like server specific information belong in the generic tool docs.

Reply:
Also, isn't the .NET SDK and CLI supposed to be intentionally platform agnostic?  To help dispel the myth that .NET is for Microsoft platforms only. I'm not sure that documenting special instructions for Azure Artifacts only helps maintain that independence.

Reply:
> AzDO does have this page on publishing packages: https://docs.microsoft.com/en-us/azure/devops/pipelines/artifacts/nuget?toc=%2Fazure%2Fdevops%2Fartifacts%2Ftoc.json&bc=%2Fazure%2Fdevops%2Fartifacts%2Fbreadcrumb%2Ftoc.json&view=azure-devops&tabs=yaml
> 
This is opposite direction (Azdo -> xxxx) of what I added here. We can't use Azdo task to push nuget to Azdo from local or Github actions.

> On the other hand, GitHub and nuget.org and are also owned by Microsoft. Should we put instructions for those NuGet servers in this doc too? How about non-Microsoft servers like MyGet, GitLabs, TeamCity, Artifactory, and who knows how many more? I might be just too idealistic, but I don't feel like server specific information belong in the generic tool docs.

Yes, you concern is valid. Still most like customer come to this doc page due to search engines so having this info relieve customer frustration. So far, MyGet, GPR (I manually tested them) and most nuget server implemenet uses `apikey`, but Azdo ignores actual value of `apikey` passed, still requires passing something (in this case `AZ`). Considering quite many customer upvoted on https://github.com/NuGet/Home/issues/7792 I believe we need to document it, I'm not aware of any other package server has a quirk like Azdo.

Reply:
> I believe we need to document it, I'm not aware of any other package server has an apikey quirk like Azdo.

I agree that we should document it, but it doesn't have to be a section in the push description page.

Reply:
> This is opposite direction (Azdo -> xxxx) of what I added here.

I copied the wrong URL, but if you looked at the table of contents on the left, it was the article immediately underneath the one I linked: [Publish a NuGet package using the command line](https://docs.microsoft.com/en-us/azure/devops/artifacts/nuget/publish?view=azure-devops). To me, that is exactly what you're documenting. How to publish a nuget package using the CLI, to an Azure Artifacts feed.

Reply:
Created link to example in Azure Artifacts examples.